docs/interactions.rst

```diff
@@ -0,0 +1,126 @@
+
+How To Make A Simple Slash Command
+----------------------------------
+
+This right here is a simple ping pong command made with Nextcord's slash feature.
+
+.. code-block:: python3
+
+    @client.slash_command()
+    async def ping(interaction):
+        await interaction.response.send_message("Pong!")
+        
+The way it works is that you use the :meth:`~Client.slash_command` function to interact with discord's application commands endpoint.
+
+``guild_ids`` is used to limit the guilds that the slash command is available to. This is also useful for testing, as global slash commands can take up to an hour to register.
+
+Example:
+
+.. code-block:: python3
+      
+    @client.slash_command(guild_ids=[id1, id2])
+
+How To Use Sub-Commands
+-----------------------
+
+The way sub-commands work is that you make a normal slash command that will never be called, and then make the sub-commands and have them do the work of real slash commands. There is no difference with slash commands and sub-commands. The only thing you will need to change is functions.
+
+As shown in the demonstration below you make a main slash command or a dummy slash command and build sub-commands off it.
+
+.. code-block:: python3
+
+        @client.slash_command(guild_ids=[...])  # Making the command and limiting the guilds
+        async def my_main_command(interaction):  
+            ...
+
+
+        @my_main_command.subcommand()  # Identifying The Sub-Command
+        async def subcommand_one(interaction):  # Making The Sub Command Name And Passing Through Interaction
+        await interaction.response.send_message(
+            "This is subcommand 1!"
+            )  # Sending A Response
+
+
+        # Identifying The Sub-Command And Adding A Descripton
+        @my_main_command.subcommand(description="Aha! Another subcommand")  
+        async def subcommand_two(interaction: Interaction):  # Passing in interaction
+
+        await interaction.response.send_message(
+            "This is subcommand 2!"
+            )       # Responding With The Args/Fields
+        
+Fields And Requirements
+-----------------------
+Fields are meant to facilitate an easier way to fill info, letting people using your slash command get a different response for each answer.
+
+Nextcord's implementation of slash commands has fields and is very simple. in the example below is a field.
+
+.. code-block:: python3
+     
+     @client.slash_command()
+     async def help(
+         interaction: Interaction,
+         setting: str = SlashOption(name="settings", description="Configure Your Settings")
+     ):
+         if setting == "music":
+             await interaction.response.send_message("Sorry we don't have PyNaCl installed currently")
+         elif setting == "moderation":
+             await interaction.response.send_message("Moderation?")
+         else:
+             await interaction.response.send_message("Odd, I don't know that setting")
```

Comment:
Surely this would be a better place to add options into the example, so that music and moderation are the only options given to the user? This would ensure people are using the right tools from the job right away, rather than learning about them much later when they already have code with unnecessary handlers like this in production.